app/validators/changed_validator.rb

```diff
@@ -0,0 +1,8 @@
+class ChangedValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    return if record.action != 'update'
+    return if value&.valid? && value&.saved_changes.present?
+
+    record.errors[attribute] << (options[:message] || "Record didn't change")
```

Comment:
ActiveModel::Dirty shold handle this, and AR should not trigger any save/update action, 
why do we need this validation?